packages/vue-components/src/Submenu.vue

```diff
@@ -0,0 +1,177 @@
+<template>
+  <li ref="submenu" :class="classes">
+    <slot name="button">
+      <a
+        class="submenu-toggle"
+        role="button"
+        :class="{disabled: disabled}"
+      >
+        <slot name="_header">
+          <slot name="header"></slot>
+        </slot>
+      </a>
+    </slot>
+    <slot name="dropdown-menu">
+      <ul class="dropdown-menu">
+        <slot></slot>
+      </ul>
+    </slot>
+  </li>
+</template>
+
+<script>
+import { toBoolean } from './utils/utils';
+import $ from './utils/NodeList';
+import positionSubmenu from './utils/submenu';
+
+export default {
+  props: {
+    show: {
+      type: Boolean,
+      default: false,
+    },
+    'class': null,
+    addClass: {
+      type: String,
+      default: '',
+    },
+  },
+  data() {
+    return {
+      dropright: true,
+      dropleft: false,
+    };
+  },
+  computed: {
+    classes() {
+      return [
+        this.class, this.addClass, 'dropdown-submenu',
+        { 'dropright': this.dropright, 'dropleft': this.dropleft },
+      ];
+    },
+    disabledBool() {
+      return toBoolean(this.disabled);
+    },
+    showBool() {
+      return toBoolean(this.show);
+    },
+    slots() {
+      return this.$slots.default;
+    },
+  },
+  methods: {
+    hideSubmenu() {
+      this.show = false;
```

Comment:
let's use `data()` for properties that could change

Reply:
Updated :)